plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -0,0 +1,235 @@
+
+import traceback
+
+try:
+    import gitlab
+    HAS_PY_GITLAB = True
+except ImportError:
+    GITLAB_IMP_ERR = traceback.format_exc()
+    HAS_PY_GITLAB = False
+
+
+class GitLabGroup(object):
+    def __init__(self, module, gl):
+        self._module = module
+        self._gitlab = gl
+
+    # get user id if the user exists
+    def get_user_id(self, gitlab_user):
+        user_exists = self._gitlab.users.list(username=gitlab_user)
+        if user_exists:
+            return user_exists[0].id
+
+    # get group id if group exists
+    def get_group_id(self, gitlab_group):
+        group_exists = self._gitlab.groups.list(search=gitlab_group)
+        if group_exists:
+            return group_exists[0].id
+
+    # check if the user is a member of the group
+    def is_user_a_member(self, gitlab_group_id, gitlab_user_id):
+        group = self._gitlab.groups.get(gitlab_group_id)
+        members = group.members.list()
+        for member in members:
+            if member.id == gitlab_user_id:
+                return True
+        return False
+
+    # add user to a group
+    def add_member_to_group(self, gitlab_user_id, gitlab_group_id, access_level):
+        try:
+            group = self._gitlab.groups.get(gitlab_group_id)
+            add_member = group.members.create(
+                {'user_id': gitlab_user_id, 'access_level': access_level})
+
+            if add_member:
+                return add_member.username
+
+        except (gitlab.exceptions.GitlabCreateError) as e:
+            self._module.fail_json(
+                msg="Failed to add member to the Group, Group ID %s: %s" % (gitlab_group_id, e))
+
+    # remove user from a group
+    def remove_user_from_group(self, gitlab_user_id, gitlab_group_id):
+        try:
+            group = self._gitlab.groups.get(gitlab_group_id)
+            group.members.delete(gitlab_user_id)
+
+        except (gitlab.exceptions.GitlabDeleteError) as e:
+            self._module.fail_json(
+                msg="Failed to remove member from GitLab group, ID %s: %s" % (gitlab_group_id, e))
+
+
+def main():
+    argument_spec = basic_auth_argument_spec()
+    argument_spec.update(dict(
+        api_token=dict(type='str', required=True, no_log=True),
+        gitlab_group=dict(type='str', required=True),
+        gitlab_user=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        access_level=dict(type='str', required=False, choices=['guest', 'reporter', 'developer', 'maintainer', 'owner'])
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['api_username', 'api_token'],
+            ['api_password', 'api_token'],
+        ],
+        required_together=[
+            ['api_username', 'api_password'],
+        ],
+        required_one_of=[
+            ['api_username', 'api_token']
+        ],
```

Comment:
```suggestion
        ],
        required_if=[
            ['state', 'present', 'access_level'],
        ],
```

Reply:
I tried this as suggested but I got the following error while I specified a value for the parameter, access_level

` "msg": "state is present but any of the following are missing: a, c, c, e, s, s, _, l, e, v, e, l"`


So, I changed it as below and it worked fine when I tested the module.

```
required_if=[
            ['state', 'present', ['access_level']],
        ],
```

For now, this adds the member to the group if they don't exist but it doesn't update the access level when they are already a member in the group.

I could think of updating the access level for an existing member if that's something we are looking for to handle in this module.

Reply:
@felixfontein , 

kinda like the idea of updating the access level for users when they are already members in the group. So, I modified the module based on that.

It should now add the user as a member in the group if they aren't, update the access level for the user if they are already members in the group, and remove the user when state is absent.

Reply:
Sorry, I totally forgot about the third element being a list!